test/e2e/auth/service_accounts.go

```diff
@@ -519,6 +520,116 @@ var _ = SIGDescribe("ServiceAccounts", func() {
 			framework.Failf("Unexpected error: %v\n%s", err, logs)
 		}
 	})
+
+	ginkgo.It("should support OIDC discovery of service account issuer [Feature:ServiceAccountIssuerDiscovery]", func() {
+
+		// Allow the test pod access to the OIDC discovery non-resource URLs.
+		clusterRoleName := "oidc-discovery"
+		if _, err := f.ClientSet.RbacV1().ClusterRoles().Create(&rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: clusterRoleName,
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					Verbs: []string{"get"},
+					NonResourceURLs: []string{
+						"/.well-known/openid-configuration",
+						"/jwks",
+					},
+				},
+			},
+		}); err != nil && !apierrors.IsAlreadyExists(err) {
+			framework.Failf("Unexpected err creating ClusterRole %s: %v", clusterRoleName, err)
+		}
+		crbName := fmt.Sprintf("%s-oidc-discovery", f.Namespace.Name)
+		if _, err := f.ClientSet.RbacV1().ClusterRoleBindings().Create(&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crbName,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      rbacv1.ServiceAccountKind,
+					APIGroup:  "",
+					Name:      "default",
+					Namespace: f.Namespace.Name,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				Name:     clusterRoleName,
+				APIGroup: rbacv1.GroupName,
+				Kind:     "ClusterRole",
+			},
+		}); err != nil && !apierrors.IsAlreadyExists(err) {
+			framework.Failf("Unexpected err creating ClusterRoleBinding %s: %v", crbName, err)
+		}
+
+		// Create the pod with tokens.
+		tokenPath := "/var/run/secrets/tokens"
+		tokenName := "sa-token"
+		audience := "oidc-discovery-test"
+		tenMin := int64(10 * 60)
+
+		pod := &v1.Pod{
```

Comment:
creating a new image and pod to test this surprised me... is there a reason not to do one of the following instead:
* an integration test
* interrogating the discovery docs / jwks from the e2e test code instead of from inside a pod
* building the capability into [agnhost](https://github.com/kubernetes/kubernetes/tree/master/test/images/agnhost) instead of introducing a new image

Reply:
Updated to use agnhost.